test/unit/test_ranges.py

```diff
@@ -0,0 +1,223 @@
+        spawn(wsgi.server, sproxyd_socket, self.sproxyd_mock)
+
+        self.sproxyd_path = '/proxy/chord'
+        conf = {
+            'swift_dir': self.swift_dir,
+            'devices': self.swift_dir,
+            'mount_check': 'false',
+            # Scality sproxyd config options
+            'sproxyd_host': '%s:%d' % sproxyd_socket.getsockname(),
+            'sproxyd_path': self.sproxyd_path,
+        }
+
+        # Hashes for the ring hashing algorithm when determining data placement
+        swift.common.utils.HASH_PATH_SUFFIX = '66a3d6b56c1f479c8b4e70ab5c2000f5'
+        swift.common.utils.HASH_PATH_PREFIX = 'changeme'
+
+        # Setup account server
+        account_path = os.path.join(self.swift_dir, 'account.ring.gz')
+        account_socket = listen(bind_address)
+        self.setup_ring(account_path, port=account_socket.getsockname()[1])
+        account_server = swift.account.server.AccountController(conf)
+
+        # Setup container server
+        container_path = os.path.join(self.swift_dir, 'container.ring.gz')
+        container_socket = listen(bind_address)
+        self.setup_ring(container_path, port=container_socket.getsockname()[1])
+        container_server = swift.container.server.ContainerController(conf)
+
+        # Setup object server
+        obj_ring_path = os.path.join(self.swift_dir, 'object.ring.gz')
+        obj_ring_socket = listen(bind_address)
+        self.setup_ring(obj_ring_path, port=obj_ring_socket.getsockname()[1])
+        object_server = swift_scality_backend.server.ObjectController(conf)
+
+        # Setup swift proxy
+        proxy_socket = listen(bind_address)
+        proxy = swift.proxy.server.Application(conf)
+
+        # Start swift proxy and auxiliary components.
+        spawn(wsgi.server, account_socket, account_server)
+        spawn(wsgi.server, container_socket, container_server)
+        spawn(wsgi.server, obj_ring_socket, object_server)
+        spawn(wsgi.server, proxy_socket, proxy)
+
+        # Create test account
+        account = 'a'
+        partition, acc_nodes = proxy.account_ring.get_nodes(account)
+        acc_node = acc_nodes.pop()
+
+        path = '/{device}/{partition}/{account}'.format(
+            device=acc_node['device'],
+            partition=partition,
+            account=account,
+        )
+        acc_headers = {'X-Timestamp': normalize_timestamp(time.time())}
+        acc_conn = HTTPConnection(acc_node['ip'], acc_node['port'])
+        acc_conn.request('PUT', path, headers=acc_headers)
+        r = acc_conn.getresponse()
+        self.assertEqual(r.status, 201)
+
+        # Create test container
+        self.proxy_host, self.proxy_port = proxy_socket.getsockname()
+        proxy_conn = HTTPConnection(self.proxy_host, self.proxy_port)
+        proxy_conn.request('PUT', '/v1/a/c/')
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 201)
+
+    def test_ranges(self):
+        self.check_path = '%s/a/c/o' % self.sproxyd_path
```

Comment:
Instead of having these attributes set, consider letting the sproxyd mock pass the information it got back into this code-path (by setting it on the instance), and putting assertions on those values here.